### PR TITLE
Avoid potential multiple invocations of loginWithRedirect

### PR DESCRIPTION
**Pull request by kweiberth**
Resolves https://github.com/auth0/auth0-react/issues/309

### CODE REVIEW

File: src/with-authentication-required.tsx [hunk 1]
```diff
@@ -84,8 +84,17 @@ const withAuthenticationRequired = <P extends object>(
     const {
       returnTo = defaultReturnTo,
       onRedirecting = defaultOnRedirecting,
-      loginOptions = {},
       claimCheck = (): boolean => true,
+      
+      // We used to default to empty object here w/ `loginOptions = {}`, but
```
Comment (adamjmcgrath) [inline src/with-authentication-required.tsx:89]: Could you remove this comment and instead add a regression test to [with-authentication-required.test.tsx](https://github.com/auth0/auth0-react/blob/master/__tests__/with-authentication-required.test.tsx)

Something like:

```js
it('should only call login once when parent state changes', async () => {
    mockClient.getUser.mockResolvedValue(undefined);
    const MyComponent = (): JSX.Element => <>Private</>;
    const WrappedComponent = withAuthenticationRequired(MyComponent);
    const App = ({ foo }: { foo: number }): JSX.Element => (
      <div>
        {foo}
        <Auth0Provider clientId="__test_client_id__" domain="__test_domain__">
          <WrappedComponent />
        </Auth0Provider>
      </div>
    );
    const { rerender } = render(<App foo={1} />);
    await waitFor(() =>
      expect(mockClient.loginWithRedirect).toHaveBeenCalled()
    );
    mockClient.loginWithRedirect.mockClear();
    rerender(<App foo={2} />);
    await waitFor(() =>
      expect(mockClient.loginWithRedirect).not.toHaveBeenCalled()
    );
  });
```

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]

**Review by adamjmcgrath — APPROVED** [review]

**Comment by adamjmcgrath** [comment]
Thanks @kweiberth! Will do put up a release straight after the holidays

**Comment by kweiberth** [comment]
Thank you @adamjmcgrath! And thanks for writing out the test. All sounds good, happy holidays!

**Comment by kevin19940627** [comment]
Hi there, withAuthenticationRequired seems to be invoking login multiple times with React 18 create react app. issues seem to appear when we use the new createRoot() API. is anyone experiencing the same issue?

**Pull request merged**